code/modules/mob/living/carbon/inventory.dm

```diff
@@ -355,4 +355,4 @@
 		return
 	visible_message("<span class='notice'>[src] takes [I] from [offerer]</span>", \
 					"<span class='notice'>You take [I] from [offerer]</span>")
-	put_in_hands(I)
+	put_in_hands(I, offered=TRUE)
```

Comment:
Instead of doing this offerered stuff, add your code to just before this is called
```
var/owner = I.get_loc_mob()
if(ismob(owner) && owner != src)
	// Some message saying ("oh no its too far away!")
	return // sorry, you won't get the benefit of your bad latency.
```